Samples/ObservableScreen/Sources/MultiCounterView.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+import Perception
+import SwiftUI
+import ViewEnvironment
+import Workflow
+import WorkflowSwiftUI
+
+struct MultiCounterView: View {
+    @Perception.Bindable var store: Store<MultiCounterModel>
+
+    var body: some View {
+        WithPerceptionTracking {
+            let _ = print("Evaluated MultiCounterView body")
+            VStack {
+                Text("Multi Counter Demo")
+                    .font(.title)
+
+                controls
+
+                if let maxCounter = store.maxCounter {
+                    CounterView(store: maxCounter, key: "max")
+                }
+
+                ForEach(
+                    Array(store.counters.enumerated()),
+                    id: \.element.id
+                ) { index, counter in
+                    HStack {
+                        Button {
+                            store.counterAction.send(.removeCounter(counter.info.id))
+                        } label: {
+                            Image(systemName: "xmark.circle")
+                        }
+
+                        CounterView(store: counter, key: "\(index)")
+                    }
+                    .padding(.vertical, 4)
+                }
+
+                // When showSum is false, changes to counters do not invalidate this body
+                if store.showSum {
+                    HStack {
+                        Text("Sum")
+                        Spacer()
+                        Text("\(store.counters.map(\.count).reduce(0, +))")
```

Comment:
This is interesting. WWDC counsels against doing potentially expensive calculations in view bodies (e.g. _O(n)_ string manipulations), but I'm not sure where else you could conveniently locate this kind of calculation across multiple observable models' values.

Reply:
Hmm, yeah, I'm not sure. If it's expensive it's not great in the workflow render either.